rudder-web/src/main/webapp/javascript/rudder/angular/dataSource.js

```diff
@@ -127,13 +132,47 @@ app.controller("datasourceCtrl", ['$scope', '$timeout', 'orderByFilter','$http',
 	  "enabled":false,
 	  "newHeader":{"key":"","value":""},
 	  "modifiedTimes":{
-	    "schedule":{"day":0,"hour":0,"minute":5},
-	    "updateTimeout":{"day":0,"hour":0,"minute":5},
-	    "requestTimeout":{"day":0,"hour":0,"minute":5}
+	    "schedule":{"second":0,"minute":5},
+	    "updateTimeout":{"second":0,"minute":5},
+	    "requestTimeout":{"second":0,"minute":5}
 	  },
+	  "modifiedHeaders":[],
+	  "modifiedParams":[],
 	  "isNew":true
 	};
   }
+  $scope.updateKeyName = function(){
+    if($scope.selectedDatasource.isNew){
+      if($scope.selectedDatasource.name){
+        var r=$scope.selectedDatasource.name.toLowerCase();
+        r = r.replace(new RegExp("\\s", 'g'),"_");
+        r = r.replace(new RegExp("[àáâãäå]", 'g'),"a");
+        r = r.replace(new RegExp("æ", 'g'),"ae");
+        r = r.replace(new RegExp("ç", 'g'),"c");
+        r = r.replace(new RegExp("[èéêë]", 'g'),"e");
+        r = r.replace(new RegExp("[ìíîï]", 'g'),"i");
+        r = r.replace(new RegExp("ñ", 'g'),"n");
+        r = r.replace(new RegExp("[òóôõö]", 'g'),"o");
+        r = r.replace(new RegExp("œ", 'g'),"oe");
+        r = r.replace(new RegExp("[ùúûü]", 'g'),"u");
+        r = r.replace(new RegExp("[ýÿ]", 'g'),"y");
+        r = r.replace(new RegExp("\\W", 'g'),"_");
+        $scope.selectedDatasource.id = r;
```

Comment:
Isn't their a function in JS to map UTF-8 space to ascii? (+take care of ascii space, at least this one is enumerable). 
Because, we could do nothing, it would be least surprising than here (for ex, what about the "ğ" ?
And why limit key space? It could be full UTF-8 chars (trimed), no?